include/seqan3/range/container/aligned_allocator.hpp

```diff
@@ -84,72 +90,108 @@ class aligned_allocator
     aligned_allocator& operator=(aligned_allocator &&)      = default; //!< Defaulted.
     ~aligned_allocator()                                    = default; //!< Defaulted.
 
-    //!\brief Copy constructor with different value type.
-    template <class other_value_type>
-    constexpr aligned_allocator(aligned_allocator<other_value_type, alignment> const &) noexcept
+    //!\brief Copy constructor with different value type and alignment.
+    template <class other_value_type, size_t other_alignment>
+    constexpr aligned_allocator(aligned_allocator<other_value_type, other_alignment> const &) noexcept
     {}
     //!\}
 
-    /*!\brief Allocates `n * sizeof(T)` bytes of uninitialized storage by calling std::aligned_alloc, but it is
-     * unspecified when and how this function is called.
-     * \throws Throws std::bad_alloc if allocation fails.
+    /*!\brief Allocates sufficiently large memory to hold `n` many elements of `value_type`.
+     *
+     * \param[in] n The number of elements for which to allocate the memory.
+     *
+     * \returns The pointer to the first block of allocated memory.
+     *
+     * \throws Throws std::bad_alloc if allocation fails, i.e. either the call to the throwing version of
+     *         operator new function throws or the requested memory exceeds the maximal number of elements to allocate.
+     *
+     * \details
+     *
+     * Allocates `n * sizeof(value_type)` bytes of uninitialized storage by calling
+     * [operator new](https://en.cppreference.com/w/cpp/memory/new/operator_new).
+     * If the given `alignment` is bigger than
+     * [__STDCPP_DEFAULT_NEW_ALIGNMENT__](https://en.cppreference.com/w/cpp/memory/new/align_val_t) the alignment
+     * aware operator new that takes as second argument the alignment as std::align_val_t. The allocation methods
+     * used by the global operator new functions is unspecified, e.g. std::malloc or std::aligned_alloc. Accordingly,
+     * the requirements on the alignment and the size of allocated bytes is implementation defined.
```

Comment:
Does this has any impact on the intended behaviour of this class?

Reply:
I don't understand you question. Could you be more specific? I mean this is the same problem when to use std::aligned_alloc. Depending on how operator new is implemented and depending on the platform different things are possible. So for example it might still be a problem in the future, that on an architecture that uses posix mem_align arbitrary sizes of bytes to allocate can be used, while on others this might be restricted to sizes that are a multiple of the alignment. But that is implementation defined and beyond the scope of this class. It only makes sure that it calls aligned memory, where it uses whatever `::operator new(n)` uses to align up to alignments of `__STDCPP_DEFAULT_NEW_ALIGNMENT__` and beyond that  aligns the memory depending on whatever `::operator new(n, alignment)` uses.

Reply:
Note: We call the new operator with the semantic requirements that the c++ standard specifies/demands, but be aware that users can overload any (global) new operator that might not adhere to the standard and might cause std::bad_alloc or unaligned pointer.